pkg/observer/data_stats.go

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Tetragon
+
+package observer
+
+import (
+	"github.com/cilium/tetragon/pkg/metrics/consts"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var (
+	// Define a counter metric for data event statistics
+	DataEventStats = promauto.NewCounterVec(prometheus.CounterOpts{
+		Namespace:   consts.MetricsNamespace,
+		Name:        "data_events_total",
+		Help:        "The number of data events by type. For internal use only.",
+		ConstLabels: nil,
+	}, []string{"event"})
+
+	DataEventSizeHist = promauto.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace:   consts.MetricsNamespace,
+		Name:        "data_event_stats_size",
```

Comment:
The last nit, promise - can it be just `data_event_size`? Having "stats" in the metric name might be a bit confusing about what is actually measured IMO :)
```suggestion
		Name:        "data_event_size",
```

Reply:
sure, np ;-)